distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date_Time.enso

```diff
@@ -380,18 +379,27 @@ type Date_Time
     ## Add the specified amount of time to this instant to produce a new instant.
 
        Arguments:
-       - amount: The amount of time to add to this instant.
+       - amount: The amount of time to add to this instant, either `Duration` for
+         adding time (hours, minutes, etc.), or `Period` for adding date (days,
+         months, years).
 
        > Example
          Add 15 years and 3 hours to a zoned date time.
 
              from Standard.Base import Date_Time
-             import Standard.Base.Data.Time.Duration
+             from Standard.Base.Data.Time import Duration, Period
 
-             example_plus = Date_Time.new 2020 + 15.years + (Duration.new hours=3)
-    + : Duration -> Date_Time ! Time_Error
+             example_plus = Date_Time.new 2020 + (Period.new years=15) + (Duration.new hours=3)
+    + : (Duration | Period) -> Date_Time ! Time_Error
     + self amount =
-        Panic.catch ArithmeticException (self.plus_builtin amount) (err -> Error.throw (Time_Error_Data err.getMessage))
+        case amount of
+            duration : Duration.Duration ->
+                Panic.catch ArithmeticException (self.plus_builtin duration) (err -> Error.throw (Time_Error_Data err.getMessage))
+            period : Period.Period ->
+                datetime = Time_Utils.datetime_adjust self Time_Utils.AdjustOp.PLUS period.internal_period
+                # We cannot return datetime directly here, as it is of type java.time.ZonedDateTime.
+                # But we can invoke `EnsoDate` methods on it.
+                new datetime.year datetime.month datetime.day datetime.hour datetime.minute datetime.second datetime.nanosecond datetime.zone
```

Comment:
As above, I'm not sure this kind of conversion should be needed at all.

But even if it is - we'd need to adjust the `new` constructor, because currently the way this conversion happens will equalize distinct dates during the DST timeline overlap, see the related Graal issue https://github.com/oracle/graal/issues/4918

Well, actually, maybe we should update the `new` constructor to be able to account for the ZoneOffset anyway, as currently there is no way to construct one of these dates during this overlap - the only way to get it is to construct a 2a.m. (GMT+2) instance and then shift it by 1 hour to get the 2.a.m (GMT+1) instance. But one cannot construct the latter directly.

Reply:
You mean something like adding a new `ZoneOffset` parameter to `Date_Time.new`?

Reply:
Possibly, although I'm not sure how the distinction between ZoneOffset and ZoneId plays with the Graal interop - we currently have just `EnsoTimeZone` type and I'm not sure if we can easily distinguish these. I guess maybe let's wait until the issue is resolved on the Graal side until we modify the constructor. We can always use the workaround to construct the date.